src/dowel/csv_output.py

```diff
@@ -48,7 +51,25 @@ def record(self, data, prefix=''):
                            'Did you change key sets after your first '
                            'logger.log(TabularInput)?'.format(
                                set(self._fieldnames), set(to_csv.keys())))
-
+                with open(self._filename, "r") as old:
+                    old_data = csv.DictReader(old)
+                    
+                    # Update the header
+                    self._fieldnames = set(self._longestHeader)|set(to_csv.keys())
+                    self._longestHeader = self._fieldnames
+
+                    # Change the writer with the new keys
+                    self._writer = csv.DictWriter(
```

Comment:
It would probably be best to refactor out the writer initialization (and header writing logic), instead of duplicating it.

Reply:
I agree. I have changed my implementation in my latest commit.